### DISCUSSION TITLE
Shopping Cart State Management: Server Side (Inertia Shared data) or Global State Management (Redux, Pinia, ...)

### DISCUSSION THREAD

**Discussion by sinar93** [inertiajs/inertia#1386] [Help] [answered]
Hi everyone
I apologize if this is a basic question, but I am having difficulty understanding the best approach.
I am looking to create a commerce website using Inertia and would like to implement a shopping cart feature.
I am unsure whether I should utilize Inertia's shared data feature on the server-side or if a front-end global state management system like Redux or Pinia would be more appropriate.
Can anyone provide guidance on this matter?
Thanks in advance

**Comment by sinar93** [comment, marked as answer]
So, here is Jonathan Reinink's answer on discord:
https://discord.com/channels/592327939920494592/1064795116289658880/1064872069235474582

**Reply by azerum** [reply]
Thank you for finding the answer. For future visitors, could you please quote the answer here? The link might rot in future. Also (not sure if it's correct) it seems like it the user has to join the server where the answer was posted to view the message

**Reply by sinar93** [reply]
Sorry my bad, Here is the answer:

**Jonathan Reinink** — 01/17/2023 3:11 PM
My my recommendation is to always use server-side state with Inertia. That's one of the main goals with Inertia, to avoid using client-side statement tools.
Just think of it as a classic server-side rendered application in terms of architecture — all data comes from the backend on page load.
